### DISCUSSION TITLE
Documentation Team. What would we like to see?

### DISCUSSION THREAD

**Discussion by bobgeorgethe3rd** [freesewing/freesewing#3951] [Support]
In the 2023-04-29 meeting we disucssed wanting to establish a documentation team. But do we want from this? And who wishes to be part of it?

**Goals/Ideas/Suggestions**
- People who can be on call to review documentation
- Maintaining Documentation that has been left.
- Ability to have a PR be halted from merging by a label that indicates Doc Review Required
- Possibly getting people to PR new Desgin Docummentation separately from the Code PR as to make reviewing of docs not stall a new designs merger.
- Review the exisiting Documnetation for any minor changes.
- Update and overhaul the existing Glossary used for re-hype
- Establish a new format for Patterm Docs and bring existing documentation to same format. This is not a change of language however translation will need to be considered and closely work with. 
- Creating Templates for users to use
- Overhauling the `addpatternsdocs.js` to use said templates
- Add yarn ability to `addpatterdocs.js` and `optionsdocsimages.mjs`
- Create a guide on how to use these scripts and has prefered terms like *good sides together*
- Added a created by and current maintainer Footnote for documentaion.

Generally make it easier for people to ask for documentation reviewing and easier for new and existing users to create, maintain and make new documentation.

**Interested**
- @SeaZeeZee 
- @starfetch
- @bobgeorgethe3rd

**Comment by BenJamesBen** [comment]
Regardless of how we might want a formal "documentation team" to work, I feel that the Documentation team Discord channel should be made public, similar to how the Pattern Design team and Development team channels are. 

When I think "documentation team" I think of a loose group of individuals who share an interest in discussing documentation matters. So, anyone with an interest can join the "documentation team" simply by participating. When I said there should be a "documentation team", I was mainly wanting the Discord channel to be created so there would be a dedicated place to discuss documentation matters.

But, I think other people are using the term to refer to more of an official "documentation moderation/administration team".

**Reply by bobgeorgethe3rd** [reply]
Yes the intention was always to make that channel public like #pattern-design and I am going to do that now as we have this place to discuss the rest.

**Reply by joostdecock** [reply]
I would suggest renaming the `documentation-team` on Discord to `documentation` as it sits under the `Contrinbutor Teams` heading, and other channels there (like `translation` or `pattern-design` don't have the `-team` suffix.

Furthermore, we might want to clean up/restructure things a bit of we are rethinking channels anyway. For example:
 - `newsletter-blogs` => What's the point?  It's not used nor is there a contributor team or anything like it that addresses this
 - `community` => Not sure this is the best name for what is essentially an `announcements` channel for contributors

So my suggestion would be to remove `newsletter-blogs` (because while we add channels, nobody seems to ever remove them) and also rename the `community` team to `news` or `updates` or whatever.

Apart from that, @BenJamesBen raises an interesting point about the nature of a so-called *documentation team*. Specifically about how formal such a team is established. In general, teams are made of up loose-knit individuals and anyone can join.
So in practice, people who show up/do the work, make up the team.

From my side, happy to agree on a set of labels to manage the PR stuff, but that's more of a #3952 discussion

**Reply by joostdecock** [reply]
Oh, an additional thought: We need to be a bit mindful on how we handle this *this documentation page maintained by...* situation  as we might very well end up with some sort of a land-grab situation where people tend to stick their flag on as many things as possible to boost their numbers/importance/ego.

So in practice:

- `createdBy` is IMHO a bad label because every page was created by someone, so it implies that when we start with this we immeadiatly need to have it everywhere. Which will be tricky. 
- I think `maintainedBy` and `lastUpdateBy` make more sense. 
- `lastUpdateBy` would (of course) only make sense if it can be automated, which may or may not be possible (I mean, it's obviously possible but the question is how long it would add to the build time).
- `maintainedBy` would then be something that can gradually be rolled out. With all pages that are not explicitly maintained by someone falling back to the current situation of them being maintained by *FreeSewing* whatever that means.

**Reply by bobgeorgethe3rd** [reply]
Channel has been renamed and is public.  As for the other channels I have informed @lexander2002 as they will have better insight. 

The idea is to have recognistion of who works on documentation and who likes to work on documentation and who should be thought of to review and be involved when a PR that includes documentation is made. Atm it is sort of a scramble to get our reviews in on PRs because we know if we don't get it done quickly it will be merged without the review. It's very much more that there are people who are willing to review and maintain doucmentation work that can be called on so that work can be spread outmore. With the additional labels we are discussing in the other discussion this will mitgated which is brilliant.

A lot of the ideas above are sort of pie in the sky and is stuff that it would be nice and I would love to work on if I was sure it is wanted. There is now enough of us that I feel as a group/team we can get to that goal and reach the sky and basically tick off this entire list. But it's bery much still is this wanted or is this just a dream.

The footnote I don't really have much of an opinion on, I added it because it was mentioned in the meeting, Personally I feel that it's not enitirely necessary for docs as docs is a free-for-all in terms of anybody can do and it should be and continue to be that anyone can do doc work. A team wouldn't succeed that. I feel knowledge of you maintains is more a thing that is needed for designs rather than documentation but that's a separate discussion to be had.

**Reply by l3xander** [reply]
I have started initializing Discord's new feature: new members can now choose which channels they want to join instead of it defaulting to all channels (without the need of a bot, all within Discord) and I believe existing members are able to change their settings too. I am going to make info roles (like someone's age) available through there as well and then we can leave the bot behind.
So let me know @joostdecock if you still want that channel deleted, since people can now opt out of it.

**Reply by joostdecock** [reply]
@lexander2002 I feel the `newsletter-blogs` one should indeed go. It hardly sees any use and while people can always opt-out there's something to be said for culling the herd from time to time 🙂

**Reply by BenJamesBen** [reply]
If `newsletter-blogs` gets deleted, where does discussion about newsletters and blogs go? Into `general`?

If `community` gets renamed `news` or `updates`, does discussion about community matters still go into that renamed channel? For example, a discussion about installing a widget onto freesewing.org so the public can read our Discord channels. Does that go into the renamed channel?

(I suppose that `general` is a catch-all for any FreeSewing discussion that isn't covered by existing channels.)

**Reply by luizfzs** [reply]
I don't think that the other channels rather than `documentation` is relevant for this discussion and may get us sidetracked.

Do we want to have the `documentation` channel be a focus point of anything documentation, in lieu of a documentation team, at least as an experiment?

**Comment by SeaZeeZee** [comment]
I'm a bit behindhand since I haven't seen the meeting notes, but from what I got from this discussion: 

My suggestion of a team came from the same experience Bob had, where I was trying to work on documentation and felt like I was on a deadline and rushing to review something before it would be merged. I would like the people like Bob who've done so much on the (sewing) documentation to feel like they can have the space and time to do what they want to do to make documentation better. 

It seems like the meeting resulted in some good ideas on how we can create that space for folks to do the documentation work they want to do without feeling rushed or unappreciated.

**Reply by BenJamesBen** [reply]
As one of the first goals, let's try to nail down the process/workflow for PRs and documentation review.

- What should the process be for the first steps after a documentation-affecting PR is filed?
- I've been adding a message to the #documentation channel, but that might make the channel tedious to read for people casually interested in documentation discussion.
- Should individuals be added to requested code reviewers for the PR in GitHub? If so, which individuals?
- Should individuals be tagged in a comment in the PR? If so, which individuals?
- Should it be up to individual doc team members to seak out newly-filed PRs to review? Perhaps via searches or using labels?

**Comment by joostdecock** [comment]
Closing to increase visibility of active discussions
